Save 2 seconds by clicking redirect link directly in a test

cbliard opened the pull request:
It can be done because this has been fixed in 8fe4a222148.

And fix the redirection not working properly if the link is clicked. Fixed by explicitly setting `window.href`.

cbliard committed:
Save 2 seconds by clicking redirect link directly in a test

It can be done because this has been fixed in 8fe4a222148.

And fix the redirection not working properly if the link is clicked.
Fixed by explicitly setting `window.href`.

as-op commented on frontend/src/stimulus/controllers/dynamic/job-status-polling.controller.ts:
```diff
@@ -40,14 +40,18 @@ export default class JobStatusPollingController extends Controller<HTMLElement>
-  redirectClick(_:Event) {
-    this.userInteraction = true;
+  redirectClick(event:PointerEvent) {
+    this.followLink(event.target as HTMLLinkElement);
```
Before the change the actual navigating was done by the browser. 
As we now set windows.href ourselves, i think it's best to ensure stopping the browser one via the event.

```suggestion
    event.preventDefault();
    this.followLink(event.target as HTMLLinkElement);
```

cbliard replied:
Thanks, I committed your suggestion.

I think the behavior when clicking is intercepted by turbo-something, and it tries to replace the turbo-frames from the current page with the turbo-frames from the response, which in turn totally skips the redirection which normally happens. Not entirely sure though.

as-op reviewed (changes requested):
👍  Thank you for following up on this!

cbliard committed:
Handle event ourselves when clicking the redirect link

As we set `window.href` ourselves, it's best to ensure stopping the browser navigation via the event

Co-authored-by: Andrej Sandorf <77627197+as-op@users.noreply.github.com>

as-op reviewed (approved):
👍
